service-base/src/main/java/org/eclipse/hono/service/resourcelimits/ResourceLimitChecks.java

```diff
@@ -86,4 +86,20 @@
      *         if the check could not be performed.
      */
     Future<Boolean> isMessageLimitReached(TenantObject tenantObject, long payloadSize, SpanContext spanContext);
+
+    /**
+     * Checks if the maximum limit of device connection duration configured for a tenant
+     * have been reached.
+     *
+     * @param tenantObject The tenant configuration to check the limit against.
+     * @param spanContext The currently active OpenTracing span context that is used to
+     *                    trace the limits verification or {@code null}
+     *                    if no span is currently active.
+     * @return A future indicating the outcome of the check.
+     *         <p>
+     *         The future will be failed with a {@link ServiceInvocationException}
+     *         if the check could not be performed.
+     * @throws NullPointerException if the tenant object is null.
+     */
+    Future<Boolean> isConnectionDurationLimitReached(TenantObject tenantObject, SpanContext spanContext);
```

Comment:
this should also have a default implementation in order to not break backward compatibility ...

Reply:
I will add it.